### Load Sample Dataset Iris Dataset

In [2]:
from sklearn.datasets import load_iris

iris = load_iris()

X = iris.data
y = iris.target

### Splitting Dataset: Training & Testing Set

In [3]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, 
                                                 y, 
                                                 test_size = 0.4, 
                                                 random_state =1)

print(f'X train: {X_train.shape}')
print(f'X test: {X_test.shape}')
print(f'y train: {y_train.shape}')
print(f'y test: {y_test.shape}')

X train: (90, 4)
X test: (60, 4)
y train: (90,)
y test: (60,)


#### Mengaktifkan package dan syntax untuk mengubah skala data

In [ ]:
##Feature Scaling
#from sklearn.preprocessing import StandardScaler  
#scaler = StandardScaler()  
#scaler.fit(x_train)

#x_train = scaler.transform(x_train)  
#x_test = scaler.transform(x_test)

### Training Model

* Pada Scikit Learn, model machine learning dibentuk dari class yang dikenal dengan istilah **estimator**
* Setiap estimator akan mengimplementasikan dua method utama yaitu `fit()` dan `predict()`
* Method `fit()` digunakan untuk melakukan training model.
* Method `predict()` digunakan untuk melakukan estimasi/prediksi dengan memanfaatkan trained model.

In [4]:
from sklearn.neighbors import KNeighborsClassifier

model = KNeighborsClassifier(n_neighbors = 3)
model.fit(X_train, y_train)

KNeighborsClassifier(n_neighbors=3)

*K-nearest neighbors* atau KNN adalah algoritma yang berfungsi untuk melakukan **klasifikasi** suatu data berdasarkan data pembelajaran (train data sets), yang diambil dari k tetangga terdekatnya (nearest neighbors). Pembentukan model ini dengan fungsi KNeighborsClassifier yang membutuhkan satu parameter yaitu spesifikasi jumlah neighbors.

### Evaluasi Model

In [5]:
from sklearn.metrics import accuracy_score

y_pred = model.predict(X_test)
acc = accuracy_score(y_test, y_pred)
print(f'Accuracy : {acc}')

Accuracy : 0.9833333333333333


Nilai akurasi dari model sebesar 98.3% yang artinya sangat baik.

### Pemanfaatan Trained Model

Kemudian dilakukan prediksi terhadap data baru menggunakan model yang didapat.

In [6]:
data_baru = [[5, 5, 3, 2]
            ,[2, 4, 3, 5]]

preds = model.predict(data_baru)
preds

array([1, 2])

In [7]:
pred_species = [iris.target_names[p] for p in preds]
print(f'Hasil Prediksi : {pred_species}')

Hasil Prediksi : ['versicolor', 'virginica']


### Dump & Load Trained Model

#### Dumping Model Machine Learning menjadi file `joblib`

In [8]:
import joblib
joblib.dump(model, 'iris_classifier_knn.joblib')

['iris_classifier_knn.joblib']

`joblib` adalah pustaka python yang digunakan untuk menyimpan dan mengimpor model mesin ke dalam file.

#### Loading Model Machine Learning dari file `joblib`

In [9]:
production_model = joblib.load('iris_classifier_knn.joblib')